x-pack/legacy/plugins/lens/public/editor_frame_plugin/editor_frame/state_management.ts

```diff
@@ -46,6 +46,12 @@ export type Action =
       type: 'UPDATE_VISUALIZATION_STATE';
       newState: unknown;
     }
+  | {
+      type: 'UPDATE_LAYER';
+      layerId: string;
+      datasourceId: string;
+      reducer: (state: unknown, layerId: string) => unknown;
```

Comment:
I don't like the name `reducer` here, the equivalent in React class components is called an `updater`: https://reactjs.org/docs/react-component.html#setstate

Reply:
Renamed it to updater, but I'm also not 100% happy with that name, because the signature of the function is somewhere in the middle between reducer (takes old state and action, returns new state), and updater (takes old state, returns new state). I think it's fine though, updater is better than reducer.